docs/specs/sequences/endpoint_reconfiguration.md

```diff
@@ -0,0 +1,150 @@
+[**UDMI**](../../../) / [**Docs**](../../) / [**Specs**](../) / [**Sequences**](./) / [Endpoint Reconfiguration](#)
+
+# Endpoint Reconfiguration
+
+Endpoint reconfiguration is the reconfiguration of the UDMI endpoint through the
+UDMI blob delivery mechanisms via UDMI config messages.
+
+The [endpoint configuration blob](https://github.com/faucetsdn/udmi/blob/master/tests/configuration_endpoint.tests/simple.json) is a JSON object defined by
+[configuration_endpoint.json](https://faucetsdn.github.io/udmi/gencode/docs/configuration_endpoint.html), which is base64 encoded in the config message.
+
+
+## Tests
+
+### Valid Endpoint (Successful Reconfiguration)
+
+**Notes**
+- `<ENDPOINT>` is a **base64** encoded endpoint object
+- `blobset.blobs_iot_endpoint_config` block is only present in a device's state messages when it is present in the last received config message
+
+```mermaid
+%%{wrap}%%
+sequenceDiagram
+    autonumber
+    participant D as Device
+    participant E as Original Endpoint
+    participant E' as New Endpoint
+    E->>D:CONFIG MESSAGE<br>timestamp = t0<br/>blobset.blobs._iot_endpoint_config = null (unset)
+    D->>E:STATE MESSAGE<br/>system.last_update = t0<br/>blobset.blobs._iot_endpoint_config = null (unset)
```

Comment:
These first two messages really are just "quiescent state" before the sequence, but actually aren't part of the sequence itself, so I'm not sure if they should be included (just because "more" is often worse than "elegant" :-).  If they follow the if-and-only-if guidelines then this is implied (so doesn't need to be explicitly stated), and I'm not sure there'd be much confusion...